### PR TITLE
Feature second refactoring

### PR DESCRIPTION
**Pull request by R-Abbasi**
This pr (second refactoring) is going to include:

- code cleanups
- code improvements
- documentation
- code formatting

### CODE REVIEW

**Review by AKMaily — COMMENTED** [review]
The version created seems to not use the calibration correctly anymore. All Scopes connected are showing ADC Counts, calibrated Scopes and not calibrated Scopes. 
image

File: src/handler.cpp [hunk 1]
```diff
@@ -1,64 +1,68 @@
...
 #include <implot.h>
-
-
 #include "popups.hpp"
 #include "get_from_github.hpp"
-
 #include "../imgui-stdlib/imgui_stdlib.h"
 
-std::vector<AxisInfo> getDeviceInfos() {
+struct AxisInfo {
```
[the remark this answers is not in the document]
Comment (R-Abbasi) [reply src/handler.cpp:9]: My scopes are not calibrated hence I can't be sure about the expected output.
Tested [this version](https://github.com/skunkforce/OmniView/actions/runs/10884351965) with my scopes. the output shows ADC Counts too. 

That struct `AxisInfo` is used merely in that source file not other places.

[the remark this answers is not in the document]
Comment (AKMaily) [reply src/handler.cpp:9]: > My scopes are not calibrated hence I can't be sure about the expected output. Tested [this version](https://github.com/skunkforce/OmniView/actions/runs/10884351965) with my scopes. The output shows ADC Counts too.

The egu that NP defined describes the measured unit. For calibrated scopes it is volt, for a different devices it maybe kelvin, celcius or a different physical unit. 
Currently it seems to always be set to ADC_Counts, which was not the case in older versions. 

> That struct `AxisInfo` is used merely in that source file not other places.

Shouldn't the declaration of the struct be in the hpp and the implementation in the cpp?

[the remark this answers is not in the document]
Comment (R-Abbasi) [reply src/handler.cpp:9]: It sounds possible to make the changes, this time step-by-step, and send you the executable to test. That calibration part is still a little vague for me although its code is not complex, anyway. 

That struct is very simple; it just has a single ctor and nothing more. To me, it's fine to have it that way in the source file.

**Comment by AKMaily** [comment]
This version now works, thank you

**Pull request merged**